core/src/Revolution/Processors/Model/GetListProcessor.php

```diff
@@ -187,6 +190,42 @@ public function getSortClassKey()
         return $this->classKey;
     }
 
+    /**
+     * Adds additional sortby criteria for grouping grids when the column being sorted is different than the one being grouped.
+     * Grouping is handled internally by Ext JS, so we do not (and should not) use groupby criteria in the query.
+     *
+     * @param xPDOQuery $c A reference to the current query being built
+     * @param string $sortBy The data index of the selected sorting column
+     * @param string $groupBy The data index of the selected grouping column
+     * @param string $groupKey The grouping column's fully qualified SQL column name
+     * @param string $gridCategory An identifier used when additional grid category-specific condition(s) are required
+     *
+     * @return void
+     */
+    public function setGroupSort(xPDOQuery &$c, string $sortBy, string $groupBy, string $groupKey, string $gridCategory = '')
+    {
+        /*
+            When group sort and column sort are the same data index, sort the groups
+            based on the current column sort direction. Otherwise, add an initial sortby
+            to specify the group sort; the secondary (sorting within the groups) is subsequently
+            added later in the getData method.
+        */
+        switch ($gridCategory) {
+            case 'usergroup-acl':
```

Comment:
Should this logic be in the generic GetListProcessor or something ACL related?

Reply:
I chose the placement of the logic based on looking to support other grouping grids down the road (e.g., the Settings one [which could use some work as mentioned in the opening comments], or any Extras that might make use of that type of grid). That said, let me know if you think there's a more appropriate place; perhaps a new processor extending the main GetListProcessor.

Reply:
@Mark-H - Hey, it's been a while since you made this comment; had a chance to consider my response? Thx ;-)

Reply:
If there needs to be an extra hook in the base getlist processor I don't have objections to that, but I do still think it's kinda weird to have a switch with values for specific implementations in a base class. That just seems like it's not architected properly.

Reply:
I agree it's not ideal; honestly there's a fair amount of wonkiness in the arch of the processors in general (e.g., there's inconsistency in how get data is handled [ two ways that I've come across ]) that makes it difficult to keep things DRY. 

I can take a look at adding an intermediate class if you prefer.

Reply:
@smg6511 I don't think it needs additional class, but just an additional method in the processor.

I'd add this to the model processor:
```php
public function getSecondaryGroupCondition(string $sortBy, string $groupBy, string $groupKey)
{
    return false;
}
```

and override the method where needed, so in the `AccessNamespace/GetList` you'd do:
```php
public function getSecondaryGroupCondition(string $sortBy, string $groupBy, string $groupKey)
{
    return $sortBy === 'authority' && $groupBy === 'role_display';
}
```

With this, you should be able to also drop the `$gridCategory` param.

Reply:
@theboxer - John, appreciate the feedback. Why didn't I think of that? ;-) Will incorporate and post later tonight...

Reply:
@theboxer @Mark-H - Ok guys, this should be ready to go ... please pull down the changes and take a look again when you get a chance.